### PR TITLE
feat: Typography and components for editorial content

### PR DESCRIPTION
**Pull request by uxengine**
Looking especially for feedback regarding editor interoperability.

Not addressed yet:
- Breakout behavior (left-only or both sides)
- Title image
- Author component for credits
- color for formats

screen shot 2017-11-13 at 16 32 04
screen shot 2017-11-13 at 16 41 21
screen shot 2017-11-13 at 16 32 22
screen shot 2017-11-13 at 16 32 29
screen shot 2017-11-13 at 16 32 33
screen shot 2017-11-13 at 16 32 43
screen shot 2017-11-13 at 16 32 59
screen shot 2017-11-13 at 16 33 08

### CODE REVIEW

File: src/components/Typography/index.js [hunk 1]
```diff
@@ -64,3 +64,108 @@ export const Interaction = {
...
+const eBoxTitleRule = css(styles.editorialBoxTitle)
+const eBoxTextRule = css(styles.editorialBoxText)
+const eAuthorLinkRule = css(styles.editorialAuthorLink)
+const eMark = css(styles.editorialMark)
+const eCaption = css(styles.editorialCaption)
+const eByline = css(styles.editorialByline)
+
+const getEditorialHeadlineRule = (type) => {
+  switch(type) {
+    case 'meta': return css(styles.editorialHeadlineMeta)
```
Comment (tpreusse) [inline src/components/Typography/index.js]: `editorialHeadlineMeta` doesn't make sense—lets make a Interaction.Headline with this style.

File: src/components/Typography/docs.md [hunk 2]
```diff
@@ -46,63 +46,227 @@ import {colors: {text}, fontStyles: {serifRegular21}} from '@project-r/styleguid
...
+  </Editorial.BoxText>
+</div>
+```
+
+### Caption and Byline
+
+These are used by the [Figure](/figure) component.
+
+```react
+<figure>
```
Comment (tpreusse) [inline src/components/Typography/docs.md]: Probably should be `<Figure>` and `<Image>`

File: src/components/Figure/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { css } from 'glamor'
+import { mUp } from '../../theme/mediaQueries'
+
+const styles = {
+  figure: css({
+    margin: '0 0 15px 0',
+    padding: 0,
+    '&[role=group]': {
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Figure/index.js]: Done.

File: src/components/Figure/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      display: 'block',
+      [mUp]: {
+        display: 'flex',
+        flexWrap: 'wrap'
+      }
+    },
+    '[role="group"] > &': {
+      [mUp]: {
+        margin: '0 7.5px'
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Figure/index.js]: Done.

File: src/components/Figure/index.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    },
+    '[role="group"] > &:last-of-type': {
+      marginRight: 0
+    }
+  })
+}
+
+export const Figure = ({ children, attributes, role, ...props }) => {
+  return (
+    <figure {...attributes} {...props} {...styles.figure} role={role}>
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Figure/index.js]: Done.

File: src/components/Figure/index.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  role: PropTypes.string,
+  type: PropTypes.string
+}
+
+Figure.defaultProps = {
+  type: 'Figure'
+}
+
+export const FigureGroup = ({ children, attributes, ...props }) => {
+  const numSiblings = children.filter(c => c.props.type === 'Figure').length
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Figure/index.js]: Done, also updated docs.md accordingly.

File: src/components/Figure/index.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  return (
+    <Figure
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Figure/index.js]: Done.

File: src/components/Figure/index.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+      {...attributes}
+      {...props}
+      {...styles.figure}
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Figure/index.js]: Done.

File: src/components/PullQuote/index.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  let containerStyle = { ...styles.container, textAlign: textAlign }
+  if (textAlign === 'center') {
+    containerStyle.margin = '0 auto'
+  }
+  if (maxWidth) {
+    containerStyle.maxWidth = maxWidth
+  }
+
+  return (
+    <blockquote {...attributes} {...css(containerStyle)}>
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/PullQuote/index.js]: Done.

File: src/components/PullQuote/index.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+const PullQuote = ({
+  children,
+  attributes,
+  source,
+  isQuoted = true,
+  textAlign = 'inherit',
+  maxWidth,
+  imageSrc,
+  ...props
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/PullQuote/index.js]: Done.

File: src/components/PullQuote/index.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+  })
+}
+
+const PullQuote = ({
+  children,
+  attributes,
+  source,
+  isQuoted = true,
+  textAlign = 'inherit',
+  maxWidth,
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/PullQuote/index.js]: Done.

File: src/components/PullQuote/index.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+      margin: '0 15px 15px 0',
+      width: '155px'
+    }
+  })
+}
+
+const PullQuote = ({
+  children,
+  attributes,
+  source,
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/PullQuote/index.js]: Done.

File: src/components/Typography/index.js [hunk 13]
```diff
@@ -64,3 +64,108 @@ export const Interaction = {
...
+    <p {...attributes} {...props} {...eBoxTextRule}>
+      {children}
+    </p>
+  ),
+  AuthorLink: ({ children, attributes, ...props }) => (
+    <a {...attributes} {...props} {...eAuthorLinkRule}>
+      {children}
+    </a>
+  ),
+  Mark: ({ children, attributes, ...props }) => (
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Typography/index.js]: Done.

File: src/components/Typography/index.js [hunk 14]
```diff
@@ -64,3 +64,108 @@ export const Interaction = {
...
+
+const getEditorialHeadlineRule = (type) => {
+  switch(type) {
+    case 'meta': return css(styles.editorialHeadlineMeta)
+    default: return css(styles.editorialHeadline)
+  }
+}
+
+// attributes are piped through for publikator editor support.
+export const Editorial = {
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Typography/index.js]: Done.

File: src/components/Typography/index.js [hunk 15]
```diff
@@ -64,3 +64,108 @@ export const Interaction = {
...
+    <a {...attributes} {...props} {...eAuthorLinkRule}>
+      {children}
+    </a>
+  ),
+  Mark: ({ children, attributes, ...props }) => (
+    <p {...attributes} {...props} {...eMark}>
+      {children}
+    </p>
+  ),
+  Caption: ({ children, attributes, ...props }) => (
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Typography/index.js]: Done.

File: src/components/Typography/docs.md (hunk 2 above)
Comment (uxengine) [reply src/components/Typography/docs.md]: Removed.

File: src/components/Typography/index.js (hunk 1 above)
Comment (uxengine) [reply src/components/Typography/index.js]: Done.

File: src/components/Typography/docs.md [hunk 16]
```diff
@@ -46,63 +46,227 @@ import {colors: {text}, fontStyles: {serifRegular21}} from '@project-r/styleguid
...
 <div {...css(styles.serifBold36)}>The quick brown fox jumps over the lazy dog</div>
 ```
 ```react|noSource,plain
+<div {...css(styles.serifBold28)}>The quick brown fox jumps over the lazy dog</div>
+```
+```react|noSource,plain
 <div {...css(styles.serifBold24)}>The quick brown fox jumps over the lazy dog</div>
 ```
 ```react|noSource,plain
+<div {...css(styles.serifBold23)}>The quick brown fox jumps over the lazy dog</div>
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Typography/docs.md]: Done.

[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Typography/docs.md]: Done.

File: src/components/Typography/docs.md [hunk 17]
```diff
@@ -46,63 +46,227 @@ import {colors: {text}, fontStyles: {serifRegular21}} from '@project-r/styleguid
...
 <div {...css(styles.sansSerifMedium40)}>The quick brown fox jumps over</div>
 ```
 ```react|noSource,plain
 <div {...css(styles.sansSerifMedium22)}>The quick brown fox jumps over the lazy dog</div>
 ```
 ```react|noSource,plain
+<div {...css(styles.sansSerifMedium20)}>The quick brown fox jumps over the lazy dog</div>
+```
+```react|noSource,plain
+<div {...css(styles.sansSerifMedium18p5)}>The quick brown fox jumps over the lazy dog</div>
```
[the remark this answers is not in the document]
Comment (uxengine) [reply src/components/Typography/docs.md]: Changed infobox titles to use font sizes of +1px instead of +.5px (confirmed with design).

**Comment by uxengine** [comment]
Took me a while, but all comments should be addressed now.

**Comment by tpreusse** [comment]
Awesome. It will be a pleasure to integrate this into the templates. Maybe some small prop changes will be required but will find those while integrating in Publikator!

**Pull request merged**